addons/test_mail/tests/test_attachment.py

```diff
@@ -0,0 +1,39 @@
+from odoo.addons.test_mail.tests.common import TestMailCommon
+from odoo.exceptions import AccessError
+
+class TestInvite(TestMailCommon):
+
+    def test_attachment_forbid_unlink(self):
+        """Check that removing message attachments is prevented on other user's messages."""
+        test_record = self.env['mail.test.simple'].with_context(self._test_context).create({
+            'name': 'Test',
+            'email_from': 'ignasse@example.com'
+        })
+
+        user_admin, user_employee = self.env['res.users'].search([('login', 'in', ['admin', 'employee'])], order='login asc')
+        admin_attachments = self.env['ir.attachment'].with_user(user_admin).create([{
+            'type': 'binary',
+            'name': f'admin_attachment_{n}',
+            'datas': "ABC=",
+        } for n in range(2)])
+        employee_attachments = self.env['ir.attachment'].with_user(user_employee).create([{
+            'type': 'binary',
+            'name': f'employee_attachment_{n}',
+            'datas': "ABC=",
+        } for n in range(2)])
+
+        test_record.with_user(user_admin).message_post(body="Hi", attachment_ids=admin_attachments.ids)
+        test_record.with_user(user_employee).message_post(body="Hello", attachment_ids=employee_attachments.ids)
+
+        # regular users should only be able to unlink their own attachments
+        with self.assertRaises(AccessError):
+            admin_attachments[0].with_user(user_employee).unlink()
+        employee_attachments[0].with_user(user_employee).unlink()
+        admin_attachments[0].with_user(user_employee).sudo().unlink()
+        self.assertEqual(len(admin_attachments.exists()), 1)
+        self.assertEqual(len(employee_attachments.exists()), 1)
+
+        # system users should be able to delete any attachment they want
+        (employee_attachments[1] | admin_attachments[1]).with_user(user_admin).unlink()
+        self.assertEqual(len(admin_attachments.exists()), 0)
```

Comment:
(same bellow)
```suggestion
        self.assertFalse(admin_attachments.exists()
```

Reply:
I reworked the test to be easier to update. Also used listEqual on the existing attachment names this way if one isn't deleted it's easy to tell from the test output which one it is.